security-model.md

```diff
@@ -0,0 +1,111 @@
+# Security Model
+
+This security model aims to clarify the project's current security position against industry 
+recommendations and general security best practices. The information provided here should make 
+it easier for security teams from highly regulated sectors to assess and use SPO for their workloads.
+
+The SPO project strives to maintain the [least privilege principle], by trying to run with 
+the bare minimum privileges for the least amount of time possible. This approach decreases
+the attack surface and the window of opportunity in which it can be abused.
+
+Note that this is an on-going effort, and may change over time as new features are developed.
+
+## Host Isolation (a.k.a. Pod Security / Security Context)
+
+Details the security requirements to run SPO from a host isolation perspective. 
+
+### Initialisation
+
+Seccomp leverages an init container to setup an initial symlink as `root`, so it can then operate
+ as a non-root user.
+
+During initialisation it also uses the capabilities:
+
+- `CHOWN`
+- `FOWNER`
+- `FSETID`
+- `DAC_OVERRIDE`
+
+
+### Running Mode
+
+The running permissions for the three core technologies supported: 
+
+|                                  | Seccomp     | SELinux     | AppArmor    |
+|----------------------------------|:-----------:|:-----------:|:-----------:|
+|               Requires root user | No          | Yes         | Yes         |
+|                 Requires HostPID | No          | No          | Yes         |
+|  Requires "privileged container" | No          | No          | Yes         |
+|     Requires SSH Access to nodes | No          | No          | No          |
+| Access to host's mount namespace | No          | Yes         | On Demand   |
+|                 AppArmor Profile | `default`   | `default`   | `default`   |
+|                     SELinux type | `spc_t`     | `spc_t`     | `spc_t`     |
+|                  Seccomp Profile | `unconfined`| `unconfined`| `unconfined`|
```

Comment:
one thing to note (maybe this can go in a separate PR) is that when SELinux is enabled, the SPOd has a sidecar container running selinuxd. selinuxd actually runs with a `selinux.process` type and is not a privileged container. Although it does run as root. Something we could start investigating is trying to run it with a non-root user and seeing if we can make do with enabling extra capabilities.... However, a lot of the attack surface of that container  is mitigated by the custom policy we use for it.